packages/@coorpacademy-components/src/molecule/base-modal/index.js

```diff
@@ -0,0 +1,132 @@
+import React, {useCallback} from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../../atom/icon';
+import ButtonLink from '../../atom/button-link';
+import style from './style.css';
+
+const BaseModal = props => {
+  const {title, description, headerIcon, children, isOpen, footer, onClose} = props;
+
+  const Footer = useCallback(() => {
+    if (!footer) return null;
+    if (typeof footer === 'function') return footer();
+
+    const {cancelButton, confirmButton, text, isError} = footer;
+
+    if (!cancelButton && !confirmButton && !text) return null;
+
+    return (
+      <div className={style.footer}>
+        <div className={style.footerCTAWrapper}>
+          {cancelButton?.onCancel && cancelButton?.label ? (
+            <ButtonLink
+              {...{
+                customStyle: {
+                  width: '94px'
+                },
+                type: 'secondary',
+                onClick: cancelButton.onCancel,
+                label: cancelButton.label,
+                disabled: cancelButton.disabled
+              }}
+            />
+          ) : null}
+          {confirmButton?.onConfirm && confirmButton?.label ? (
+            <ButtonLink
+              {...{
+                customStyle: {
+                  width: '121px'
+                },
+                type: 'primary',
+                onClick: confirmButton.onConfirm,
+                label: confirmButton?.label,
+                disabled: confirmButton?.disabled,
+                ...(confirmButton?.iconName
+                  ? {
+                      icon: {
+                        position: 'left',
+                        faIcon: {
+                          name: confirmButton?.iconName,
+                          color: '#FFFFFF',
+                          size: 16
+                        }
+                      }
+                    }
+                  : null)
+              }}
+            />
+          ) : null}
+        </div>
+        {text ? <div className={isError && style.footerDescriptionError}>{text}</div> : null}
+      </div>
+    );
+  }, [footer]);
+
+  if (!isOpen || !title || !children) return null;
+
+  function handleOnClose(e) {
+    e?.stopPropagation();
+    onClose();
+  }
+
+  return (
+    <div className={style.modalWrapper}>
+      <div className={style.modal}>
+        <header className={style.header}>
+          {headerIcon && headerIcon.name ? (
```

Comment:
!isEmpty(headerIcon) ?

Reply:
we still need to check the `name` property, so I will replaced it with `headerIcon?.name ?` to shorten the code instead. Thank you